### PR TITLE
Add argh 1.3.1

### PR DESCRIPTION
**Pull request by a4z**
Specify library name and version:  **argh/1.3.1**

This library already existed at some point of time, but is deprecated since some time
https://conan.io/center/argh?os=&tab=configuration

So here the latest version, to get it back, since users request it, see https://github.com/adishavit/argh/issues/59

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/argh/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (uilianries) [inline recipes/argh/all/conanfile.py:2]:
```suggestion
```
Not used

File: recipes/argh/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    license = "BSD-3"
+    description = "Frustration-free command line processing"
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return os.path.join(self.source_folder, "source_subfolder")
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/argh/all/conanfile.py:20]:
```suggestion
    def validate(self):
```
https://blog.conan.io/2020/12/14/New-conan-release-1-32.html

File: recipes/argh/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("argh-{}".format(self.version), self._source_subfolder)
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("argh.h", src=os.path.join(self._source_subfolder), dst=os.path.join("include", "argh"))
```
Comment (uilianries) [inline recipes/argh/all/conanfile.py:30]:
```suggestion
        self.copy("argh.h", src=self._source_subfolder, dst=os.path.join("include", "argh"))
```

File: recipes/argh/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+from io import StringIO
```
Comment (uilianries) [inline recipes/argh/all/test_package/conanfile.py:2]:
```suggestion
```

Not used

**Review by uilianries — COMMENTED** [review]
That's a nice project which we have as legacy, it's cool to see it again and updated. Only few changes and we are good to go.

**Comment by a4z** [comment]
Thanks a lot for your improvements Uilian!

I seem to totally have missed the `validate` method, is this kind of new?

**Comment by uilianries** [comment]
> I seem to totally have missed the validate method, is this kind of new?

It's available since Conan 1.32. It's the new way for cppstd validation. For this recipe doesn't matter, but it keeps same pattern at least. In some cases we need to consider not recipe requirements only, but also its dependencies. The `validate()` runs one step before `build()` and contains the dependencies tree, which allows us validating transitive options and cppstd, for instance. Usually we needed to do it under `build()` or `configure()`, depending the situation, but now we can centralize all into `validate()`.

**Review by uilianries — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Pull request merged**